alica_engine/src/engine/parser/PlanWriter.cpp

```diff
@@ -522,7 +519,15 @@ string PlanWriter::getRelativeFileName(string file) {
         if (file.substr(file.size() - 4, 4) == ".beh" || file.substr(file.size() - 4, 4) == ".pty" ||
                 file.substr(file.size() - 4, 4) == ".pml" || file.substr(file.size() - 3, 3) == ".pp") {
             supplementary::SystemConfig* sc = supplementary::SystemConfig::getInstance();
-            string tfile = (*sc)["Alica"]->get<string>("Alica.PlanDir");
+            string tfile = (*sc)["Alica"]->get<string>("Alica.PlanDir", NULL);
+            supplementary::FileSystem::combinePaths(tfile, file);
+            if (!supplementary::FileSystem::isPathRooted(tfile)) {
+                tfile = supplementary::FileSystem::combinePaths(this->configPath, tfile);
+            }
+            ufile = tfile;
+        } else if (file.substr(file.size() - 4, 4) == ".tsk") {
+            supplementary::SystemConfig* sc = supplementary::SystemConfig::getInstance();
+            string tfile = (*sc)["Alica"]->get<string>("Alica.MiscDir", NULL);
             supplementary::FileSystem::combinePaths(tfile, file);
```

Comment:
The returned string is not used

Reply:
This function in general has unneeded variables and requires a cleanup

Reply:
yes it needs to have a cleanup. Not now though